### DISCUSSION TITLE
Go LSP Not working

### DISCUSSION THREAD

**Discussion by kyp0717** [nix-community/kickstart-nix.nvim#49] [Q&A] [answered]
Hello.  I am trying to implement LSP support for go programming.
My understanding is that In the `./nvim/ftplugin` folder, I should create the `go.lua` with the following code.  With `go.lua` in `ftplugin`, that is all I need for `nvim` to recognize `go` files so that LSP (using `gopls`) can be started and attached to the buffer.  However, this approach did not work.  What am I doing wrong?

```lua
// ./nvim/ftplugin/go.lua
local autocmd = vim.api.nvim_create_autocmd
autocmd("FileType", {
    pattern = "go",
    callback = function()
        local root_dir = vim.fs.dirname(
            vim.fs.find({ 'go.mod', 'go.work', '.git' }, { upward = true })[1]
        )
        local client = vim.lsp.start({
            name = 'gopls',
            cmd = { 'gopls' },
            root_dir = root_dir,
        })
        vim.lsp.buf_attach_client(0, client)
    end
})
```

`gopls` have already been installed in the `flake.nix`
The go filetype was recognized but the LSP server did not start.  `LspInfo` indicated that a client was not started.
Please advise.  Thanks.

**Comment by mrcjkb** [comment, marked as answer]
hey 👋 

You shouldn't create an autocommand in `ftplugin` scripts, as the scripts are sourced on filetype events.

**Reply by kyp0717** [reply]
Thank you for your respond. I have modified based on your recommendation and  it works now!
